source/patterns/@aws-solutions-constructs/aws-lambda-kinesisstreams/test/lambda-kinesisstream.test.ts

```diff
@@ -0,0 +1,495 @@
+    StreamEncryption: {
+      EncryptionType: 'KMS',
+      KeyId: 'alias/aws/kinesis'
+    }
+  });
+});
+
+test('CloudWatch Alarms are created for Kinesis Stream by default', () => {
+  const stack = new cdk.Stack();
+
+  new LambdaToKinesisStreams(stack, 'test-lambda-kinesisstreams', {
+    lambdaFunctionProps: {
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'index.handler'
+    }
+  });
+
+  expect(stack).toHaveResourceLike('AWS::CloudWatch::Alarm', {
+    Namespace: 'AWS/Kinesis',
+    MetricName: 'GetRecords.IteratorAgeMilliseconds'
+  });
+
+  expect(stack).toHaveResourceLike('AWS::CloudWatch::Alarm', {
+    Namespace: 'AWS/Kinesis',
+    MetricName: 'ReadProvisionedThroughputExceeded'
+  });
+});
+
+test('CloudWatch Alarms are not created when createCloudWatchAlarms property is set to false', () => {
+  const stack = new cdk.Stack();
+
+  new LambdaToKinesisStreams(stack, 'test-lambda-kinesisstreams', {
+    lambdaFunctionProps: {
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'index.handler'
+    },
+    createCloudWatchAlarms: false
+  });
+
+  expect(stack).toCountResources('AWS::CloudWatch::Alarm', 0);
+});
+
+test('Error is thrown when vpc is specified and existing lambda function is not associated with it', () => {
+  const stack = new Stack();
+
+  const existingLambdaObj = new lambda.Function(stack, 'test-lamba', {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_18_X,
+    handler: 'index.handler'
+  });
+
+  const existingVpc = new ec2.Vpc(stack, 'test-vpc', {});
+
+  const app = () => {
+    new LambdaToKinesisStreams(stack, 'lambda-to-sqs-stack', {
+      existingLambdaObj,
+      existingVpc
+    });
+  };
+
+  expect(app).toThrowError();
+});
+
+test('Construct uses existing Lambda Function', () => {
+  const stack = new Stack();
+
+  const existingLambdaObj = new lambda.Function(stack, 'test-lamba', {
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    runtime: lambda.Runtime.NODEJS_18_X,
+    handler: 'index.handler',
+    functionName: 'my-lambda-function'
+  });
+
+  new LambdaToKinesisStreams(stack, 'lambda-to-sqs-stack', {
+    existingLambdaObj
+  });
+
+  expect(stack).toHaveResourceLike('AWS::Lambda::Function', {
+    FunctionName: 'my-lambda-function'
+  });
+});
+
+test('Construct uses existing Kinesis Stream', () => {
+  const stack = new Stack();
+
+  const existingStreamObj = new kinesis.Stream(stack, 'test-stream', {
+    streamName: 'my-stream',
+  });
+
+  new LambdaToKinesisStreams(stack, 'test-lambda-kinesisstreams', {
+    lambdaFunctionProps: {
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: 'index.handler'
+    },
+    existingStreamObj
+  });
+
+  expect(stack).toHaveResourceLike('AWS::Kinesis::Stream', {
```

Comment:
Add a check that resource count is 1 to ensure we didn't create one anyway.

Reply:
will do!